packages/e2e-tests/helpers/new-react-devtools-panel.ts

```diff
@@ -32,6 +105,98 @@ export async function openReactDevtoolsPanel(page: Page) {
   await openReactDevtoolsPanelOld(page);
 }
 
+export async function searchComponents(page: Page, text: string) {
+  const input = getSearchInput(page);
+
+  await waitFor(async () => {
+    await expect(await input.isEnabled()).toBe(true);
+  });
+
+  await waitFor(async () => {
```

Comment:
I don't think this `waitFor()` makes any sense since we don't `expect()` anything in its body. Also we wouldn't want to run its body more than once because then it would fill the search input with multiple copies of the `text`.

Reply:
Maybe the wait-for is unnecessary. I added in case (for some reason) the search didn't yield results.

> it would fill the search input with multiple copies of the text.

This isn't correct though. This line will clear the input (if there's any text in it from previously, including a previous wait-for iteration):
```js
    await input.press("Escape"); // Clear/reset
```

Reply:
> I don't think this waitFor() makes any sense since we don't expect() anything in its body.

Just to be clear though, we do expect something–
```js
    const searchResults = page.locator('[data-test-id="ReactDevTools-SearchResults"]');
    await searchResults.waitFor();
```

The above lines will fail if a search isn't active. (The search results UI won't exist.)